README.md

```diff
@@ -1,52 +1,105 @@
 # ssh
 ![CI Testing](https://github.com/linux-system-roles/ssh/workflows/tox/badge.svg)
 
-An Ansible role for managing ssh clients.
+An Ansible role for managing ssh clients configuration.
 
 ## Requirements
 
-Any pre-requisites that may not be covered by Ansible itself or the role should
-be mentioned here. For instance, if the role uses the EC2 module, it may be a
-good idea to mention in this section that the `boto` package is required.
+This role should work on any system that provides openssh client and is
+supported by ansible. The role was tested on:
+
+ * RHEL/CentOS 6, 7, 8
+ * Fedora 32, 33
+ * Debian
+ * Ubuntu
 
 ## Role Variables
 
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
+By default, the role should not modify the system configuration and generate
+global `ssh_config` that matches OS default (the generated configuration does
+not keep comments and order of the options).
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+ * `ssh_user`:
 
-Example of setting the variables:
+By default (*null*) the role will modify the global configuration for all
+users. Other values will be interpretted as a username and the role will
+modify per-user configuration stored under `~/.ssh/config` of the given user.
+The user needs to exists before invoking this role otherwise it will fail.
 
-```yaml
-ssh_foo: "oof"
-ssh_bar: "baz"
-```
+ * `ssh_skip_defaults`:
+
+By default (`auto`), the role writes the system-wide configuration file
+`/etc/ssh/ssh_config` and keeps OS defaults defined there (*true*). This is
+automatically disabled, when a drop-in configuration file is created
+(`ssh_drop_in_name!=null`) or when per-user configuration file is created
+(`ssh_user!=null`).
+
+ * `ssh_drop_in_name`:
+
+This defines the name for the drop-in configuration file to be placed in
+system-wide drop-in directory. The name is used in the template
+defined by `__ssh_drop_in_template` (by default
```

Comment:
I'm curious that do we share the concept with the users that a variable starting with `__` is "internal" and it's something the users cannot set or are not trying to modify? (I feel just mentioning `/etc/ssh/ssh_config.d/{name}.conf` here without `__ssh_drop_in_template` might be clear enough...)

Reply:
Right. I will remove it from readme and add a note into the `contributing.md`. This is something which might be interesting for adding support for other distros/oses that might decided to use different location for drop-in directory (if we are unlucky).

Reply:
Now, it makes me wonder if we include contributing.md in the package, we should include .dev-tools, too? Otherwise, the contents of the file would be too cryptic...

As @pcahyna commented, we are removing files and dirs starting with '.' in the spec... Should we resurrect the dir?
```
rm -r $RPM_BUILD_ROOT%{_datadir}/ansible/roles/%{roleprefix}*/.[A-Za-z]*
```
Another idea is we could keep both contributing.md and .dev-tools in the upstream package and drop both from the downstream. It may add some extra burden to the packaging task.

Reply:
What would be even better is that the package only contains files that Ansible knows about, or files that are otherwise required for the packaging system, and nothing else.  However, we have already established the precedent of keeping files such as `network/contributing.md` in the package, so I suppose it is ok to keep `ssh/contributing.md` in the package as well.  That said, I do not think it is necessary to keep `.dev-tools/` in the package - I think we can safely assume that anyone looking for `.dev-tools/` is going to be a developer working on the source from github.  I'm not worried about fielding support calls from confused customers installing from rpm or galaxy and unable to find some directory called `.dev-tools/` mentioned in `contributing.md`